File: Java_Generated/Java/untitled/src/measurements/DataConsumer.java
package measurements;

import category_theory.CategoryObject;
import diagram.interfaces.IDesktop;
import general_service.Entry;
import general_service.Performer;
import general_service.interfaces.IPostSetArrow;
import general_service.interfaces.IValueSetter;
import general_service.interfaces.IValueSetterFactory;
import measurements.interfaces.IDataConsumer;
import measurements.interfaces.IMeasurements;
import measurements.time.interfaces.ITimeMeasurementConsumer;
import measurements.time.interfaces.ITimeMeasurementProvider;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class DataConsumer extends CategoryObject implements IDataConsumer, IPostSetArrow,
        ITimeMeasurementConsumer
{

    protected Performer performer = new Performer();

    ITimeMeasurementConsumer tms;

    ITimeMeasurementProvider timeMeasurement;

    protected boolean success = true;

    protected IDataConsumer dataConsumer;

    protected IValueSetterFactory valueSetterFactory;

    protected List<IMeasurements> measurements = new ArrayList<>();

    protected Object variable;

    public DataConsumer(String name, IDesktop desktop)
    {
        super(name, desktop);
        valueSetterFactory = desktop.getValueSetterFactory();
    }

    protected void init() {

    }



    @Override
    public IMeasurements[] getAllMeasurements() {
        var size = measurements.size();
        IMeasurements[] m = new IMeasurements[size];
        measurements.toArray(m);
        return m;
    }

    @Override
    public void addMeasurements(IMeasurements item)
    {
         measurements.add(item);
    }

    @Override
    public void postSetArrow() {

    }

    @Override
    public ITimeMeasurementProvider getTimeMeasutement() {
        return timeMeasurement;
    }

    @Override
    public void setTimeMeasurement(ITimeMeasurementProvider provider) {
timeMeasurement = provider;
    }

    @Override
    public double getInternalTime() {
        return 0;
    }

    protected String[] copyMap(Map<String, Entry<Object, Object>> input, Map<String, IValueSetter> output)
    {
        return mPefrformer.copyMap(input, output, valueSetterFactory);
    }


    protected measurements.Performer mPefrformer = new measurements.Performer();
}

File: src/Python/lib/measurements/data_consumer.py
from ..category_object import CategoryObject
from typing import List
from measurements import Measurements

class DataConsumer(CategoryObject): 

    def __init__(self, name, desktop):
        super().__init__(name, desktop)
        self._measurements : List[Measurements] = []
        self._dependent_measurements : List[Measurements] = []
        # self._objects : list = []

    @property
    def measurements(self) -> List[Measurements]:
        return self._measurements
    
    @property
    def dependent_measurements(self) -> List[Measurements]:
        return self._dependent_measurements
    
    def append(self, item: Measurements):
        self._measurements.append(item)
        # self._objects = []
        dependent_objects = []
        self._dependent_measurements = []
        for m in self._measurements:
            self._dependent_measurements = self._dependent_measurements.append(m)
            if issubclass(m, DataConsumer):
                dependent_objects = self.get_dependent_objects(dependent_objects)
                for dep_obj in dependent_objects:
                    if issubclass(dep_obj, Measurements) and dep_obj not in self.dependent_measurements:
                        self._dependent_measurements.append(dep_obj)
        

                
    def get_dependent_objects(self, objects_list: list = []) -> list:
        for m in self.measurements:
            if m not in objects_list:
                objects_list.append(m)
            if issubclass(m, DataConsumer):
                objects_list += m.get_dependent_objects(objects_list)
        return objects_list
             






    

